src/main/java/it/gov/pagopa/payhub/activities/service/debtposition/ionotification/IONotificationService.java

```diff
@@ -0,0 +1,112 @@
+
+@Lazy
+@Service
+@Slf4j
+public class IONotificationService {
+
+    private final DebtPositionTypeOrgService debtPositionTypeOrgService;
+    private final IONotificationFacadeService ioNotificationFacadeService;
+    private final NotificationRequestMapper notificationRequestMapper;
+    private final InstallmentOperationTypeResolver installmentOperationTypeResolver;
+
+    public IONotificationService(DebtPositionTypeOrgService debtPositionTypeOrgService, IONotificationFacadeService ioNotificationFacadeService, NotificationRequestMapper notificationRequestMapper, InstallmentOperationTypeResolver installmentOperationTypeResolver) {
+        this.debtPositionTypeOrgService = debtPositionTypeOrgService;
+        this.ioNotificationFacadeService = ioNotificationFacadeService;
+        this.notificationRequestMapper = notificationRequestMapper;
+        this.installmentOperationTypeResolver = installmentOperationTypeResolver;
+    }
+
+    public List<MessageResponseDTO> sendMessage(DebtPositionDTO debtPositionDTO, Map<String, IupdSyncStatusUpdateDTO> iupdSyncStatusUpdateDTOMap) {
+        List<MessageResponseDTO> response = new ArrayList<>();
+
+        DebtPositionDTO filteredDebtPosition = filterInstallmentsByIUD(debtPositionDTO, iupdSyncStatusUpdateDTOMap);
+
+        if (!filteredDebtPosition.getPaymentOptions().isEmpty()) {
+            DebtPositionDTO createDpDebtPosition = calculateOperationType(filteredDebtPosition);
+
+            if (!createDpDebtPosition.getPaymentOptions().isEmpty()) {
+                IONotificationDTO ioNotificationDTO = debtPositionTypeOrgService
+                        .getIONotificationDetails(debtPositionDTO.getDebtPositionTypeOrgId(), CREATE_DP);
+
+                if (ioNotificationDTO != null) {
+                    response = processNotifications(createDpDebtPosition, ioNotificationDTO);
+                }
+            }
+        }
+
+        return response;
+    }
+
+    /**
+     * Maps and Sends notifications for the given debt position.
+     */
+    private List<MessageResponseDTO> processNotifications(DebtPositionDTO debtPositionDTO, IONotificationDTO ioNotificationDTO) {
+        List<NotificationRequestDTO> notifications = notificationRequestMapper.map(
+                debtPositionDTO, ioNotificationDTO, CREATE_DP
+        );
+
+        return notifications.stream()
+                .map(ioNotificationFacadeService::sendMessage)
+                .toList();
+    }
+
+    /**
+     * Filters payment options to keep only installments with the CREATE_DP operation type.
+     */
+    private DebtPositionDTO calculateOperationType(DebtPositionDTO debtPositionDTO) {
+        debtPositionDTO.getPaymentOptions().forEach(po ->
+                po.setInstallments(po.getInstallments().stream()
+                        .filter(i -> NotificationRequestDTO.OperationTypeEnum.CREATE_DP
+                                .equals(installmentOperationTypeResolver.calculateOperationType(i)))
```

Comment:
This way you are deciding if it's a creation just considering the single installment, if you are adding an installment to an already existing DP this should be considered a CREATE operation?